packages/compass-sidebar/src/stores/store.js

```diff
@@ -23,32 +23,38 @@ store.onActivated = (appRegistry) => {
       changeInstance({
         refreshingStatus: instance.refreshingStatus,
         databasesStatus: instance.databasesStatus,
-        isRefreshing: instance.isRefreshing,
       })
     );
-  }, 100);
+  }, 300);
```

Comment:
Just want to highlight this change a bit more:

As much as I tried to make state updates granular and only update changed values super efficiently (e.g., carefully adding / removing items from the array, replacing only changed items on relevant change event, instead of doing a copy of the whole array), things like making immutable copies of arrays when one item in the array changes (required pattern to trigger redux state to update the component) was a giant bottleneck for any server that has more than a ~1k dbs / colls. This is why I'm keeping the throttle and even bumping the throttle timing in this PR a little. This still gives some UI lag in some cases, but works much better than it was working before we started work on TTFQ